server/integration/infinispan/src/main/java/org/infinispan/server/infinispan/task/DistributedServerTask.java

```diff
@@ -0,0 +1,55 @@
+package org.infinispan.server.infinispan.task;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.marshall.Marshaller;
+import org.infinispan.distexec.DistributedCallable;
+import org.infinispan.factories.GlobalComponentRegistry;
+import org.infinispan.tasks.TaskContext;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Author: Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ * Date: 1/28/16
+ * Time: 1:49 PM
+ */
+public class DistributedServerTask<T> implements Serializable, DistributedCallable<Object, Object, T> {
+   private final Map<String, ?> parameters;
+   private final String taskName;
+
+   private transient GlobalComponentRegistry componentRegistry;
+   private transient Cache<Object, Object> cache;
+
+   public DistributedServerTask(String taskName, Map<String, ?> parameters) {
```

Comment:
Just thinking... Maybe this should accept Optional of map? This would simplify this: https://github.com/infinispan/infinispan/pull/3950/files#diff-e136f12dbae088bc0d3481f744bb9c97R26